Source/WebKit/Platform/IPC/HandleMessage.h

```diff
@@ -364,18 +480,34 @@ void handleMessageSynchronous(Connection& connection, Decoder& decoder, UniqueRe
     static_assert(std::is_same_v<typename ValidationType::CompletionHandlerArguments, typename MessageType::ReplyArguments>);
     using CompletionHandlerType = typename ValidationType::CompletionHandlerType;
 
-    CompletionHandlerType completionHandler(
+    logMessage(connection, MessageType::name(), object, *arguments);
+    if constexpr (ValidationType::completionHandlerIsCelled) {
+        Ref<CompletionHandlerType> completionHandler = CompletionHandlerType::create(
         [replyEncoder = WTFMove(replyEncoder), connection = Ref { connection }] (auto&&... args) mutable {
             logReply(connection, MessageType::name(), args...);
             (replyEncoder.get() << ... << std::forward<decltype(args)>(args));
             connection->sendSyncReply(WTFMove(replyEncoder));
         });
 
-    logMessage(connection, MessageType::name(), object, *arguments);
-    if constexpr (ValidationType::expectsConnectionArgument)
-        callMemberFunction(object, function, connection, WTFMove(*arguments), WTFMove(completionHandler));
-    else
-        callMemberFunction(object, function, WTFMove(*arguments), WTFMove(completionHandler));
+        if constexpr (ValidationType::expectsConnectionArgument) {
```

Comment:
The changes to this file aren't anywhere near as significant as they seem. It's just copypasta inside `if/else` conditions representing whether we're calling Swift or C++ receivers, with the slightly different ways they need to receive their arguments.